File: day2/part1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("./input.txt")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)
	lines := make([][]string, 0)
	for scanner.Scan() {
		lines = append(lines, splitLine(scanner.Text()))
	}
	x := 0
	y := 0
	for _, v := range lines {
		val, err := strconv.Atoi(v[1])
		if err != nil {
			panic(err)
		}
		switch v[0] {
		case "forward":
			x += val
		case "down":
			y += val
		case "up":
			y -= val
		}
	}
	fmt.Println(x * y)
}

func splitLine(line string) []string {
	return strings.Split(line, " ")
}

File: day8/part1.js
const fs = require('fs')
const data = fs.readFileSync("./input.txt").toString()

const lines = data.split('\n')
console.log(lines.map(line => {
    const [values, output] = line.split(' | ')
    return output.split(' ').map(o => [2, 3, 4, 7].includes(o.length)).filter(a => a).length
}).reduce((a, b) => a + b, 0));